limesurvey/templates/deployment.yaml

```diff
@@ -62,15 +62,52 @@ spec:
               mountPath: {{ .mountPath }}
           {{- end }}
           env:
+            - name: DB_TYPE
+              {{- if eq .Values.mariadb.enabled true }}
+              value: "mysql"
+              {{- else }}
+              value: {{ .Values.externalDatabase.type }}
+              {{- end }}
             - name: DB_HOST
+              {{- if eq .Values.mariadb.enabled true }}
+              value: {{ include "limesurvey.mariadb.fullname" . }}
+              {{- else }}
               value: {{ .Values.externalDatabase.host }}
+              {{- end }}
+            - name: DB_PORT
+              {{- if eq .Values.mariadb.enabled true }}
+              value: {{ .Values.mariadb.primary.service.port | quote }}
+              {{- else }}
+              value: {{ .Values.externalDatabase.port | quote }}
+              {{- end }}
             - name: DB_PASSWORD
+              {{- if eq .Values.mariadb.enabled true }}
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.mariadb.auth.existingSecret }}
+                  name: {{ .Values.mariadb.auth.existingSecret }}
+                  {{- else }}
+                  name: {{ include "limesurvey.mariadb.fullname" . }}
+                  {{- end }}
+                  key: mariadb-password
+              {{- else }}
               valueFrom:
                 secretKeyRef:
                   name: {{ include "limesurvey.databaseSecretName" . }}
                   key: mariadb-password
+              {{- end }}
             - name: DB_USERNAME
+              {{- if eq .Values.mariadb.enabled true }}
+              value: {{ .Values.mariadb.auth.username }}
+              {{- else }}
               value: {{ .Values.externalDatabase.username }}
```

Comment:
Hmm, maybe I should wrap these values in a secret instead of directly accessing them...